lib/fuse_lowlevel.c

```diff
@@ -2333,9 +2333,21 @@ int fuse_lowlevel_notify_expire_entry(struct fuse_session *se, fuse_ino_t parent
 }
 
 int fuse_lowlevel_notify_inval_entry(struct fuse_session *se, fuse_ino_t parent,
-				     const char *name, size_t namelen)
+						 const char *name, size_t namelen)
+{
+	return fuse_lowlevel_notify_entry(se, parent, name, namelen, FUSE_LL_INVALIDATE);
+}
+
+int fuse_lowlevel_notify_expire_entry(struct fuse_session *se, fuse_ino_t parent,
+							const char *name, size_t namelen)
 {
-	return fuse_lowlevel_notify_expire_entry(se, parent, name, namelen, 0);
+	if (!se)
+		return -EINVAL;
+
+	if (!(se->conn.capable & FUSE_CAP_EXPIRE_ONLY))
```

Comment:
Do we need to do this check? What happens if we just pass the flag to the kernel?

If the kernel returns an error (which I really hope is the case) then we could keep this code simpler.

Reply:
as far as i know the kernel reports no error for backward compatibility - that's why it just fallsback to `invalidate`

Reply:
the kernel does not return any error. it just falls back to `invalidate`. thats why you need a check for it